rascal-lsp/src/main/rascal/util/LanguageServer.rsc

```diff
@@ -60,6 +63,14 @@ data LanguageService
     | lenses(LensDetector detector)
     | inlayHinter(InlayHinter hinter)
     | executor(CommandExecutor executor)
+    | definer(SingleMapper define)
+    | referrer(SingleMapper reference) // this
+    | implementer(SingleMapper implementations)
+    | disabledSummary( // tell rascal that the summary should not be run to calculate these
```

Comment:
I'm not happy with this "disabler" semantics. If someone does not want to calculate something, then they can simply decide not to calculate something. We do not need booleans for that. A summary that does not have `definitions`, does not have definitions.

Reply:
we can just have the definer, referrer and implementer, and a summary provider. All we have to do is decide what to do when we have them both. In my view the faster single functions would run first and do a fast approximation, while the summary provider would catch up sometimes and provide more accurate relations.

Reply:
it makes no sense to calculate the summary references variable with a lot of detail, only to turn the information off in the IDE.

Reply:
It should maybe be called differently, but I want to allow the user to say (which we also had in eclipse ide) I'm not providing this information in the summary, so you can save requesting it, and just send an empty reply back.

Reply:
For example, if your DSLs can't implement references via the summary, but only on demand, you want to communicate this.

Similarly for `documentation` if your summary doesn't contain it, why wait for it to finish calculating. (or why even request it).

Reply:
I want to find a different solution for this problem, if it is a problem. The current solution couples 5 different contributors, semantically, via boolean flags. That is not good. The contributors should remain independent functional callbacks and the protocol that uses them should remain "under the hood". Here things start to tangle in a way that is too ad-hoc. 

To indicate that something is not there, the design principle here is to simply not provide it. So: no callback contributed, or no relation added to the summary.

Reply:
this was rewritten after discussion.